astropy/utils/masked/tests/test_masked.py

```diff
@@ -290,6 +290,16 @@ def test_viewing(self):
         assert_array_equal(ma.unmasked, self.a.view(np.ndarray))
         assert_array_equal(ma.mask, self.m)
 
+    def test_viewing_independent_shape(self):
```

Comment:
This extra test was also triggered by your review: if I made such an effort *not* to view the mask, is that a problem? Thankfully, it turns out it is not - a view is taken in `_set_mask`. But I couldn't find any test for it, so thought I might as well add one now, since this PR touches the viewing code.